### ISSUE TITLE
Feature Request: Add Settings for Link Placement and Default Tag

### ISSUE THREAD

**Issue by AHCorn** [thesamim/TickTickSync#131]
Thank you for creating this plugin; it has perfectly met my needs. However, I have encountered two issues that I could not resolve even after reviewing the documentation and existing issues. Perhaps these features have not yet been implemented.

1.	When syncing, the plugin automatically includes Obsidian links in the tasks. Initially, I found this convenient, but as the number of tasks grows, especially with fixed date titles like those in journals, it leads to a cluttered task list in the TickTick app. Could these links possibly be added to the description instead of the task title?

2.	I sync tasks using the default #ticktick tag, but I noticed that this tag is also synced to TickTick. I feel this is unnecessary. Perhaps this tag could be changed to #obsidian during sync, it’s possible that I simply haven’t found how to configure this.

I really appreciate this plugin and am excited to see what comes next.

**Comment by thesamim** [comment]
Thanks for the feedback.

Will give it some thought.

**Comment by martin-arg** [comment]
I really like what Ahcorn proposes, and I ask if I can here the links in obsidian to ticktick and necessary or a feature?

**Comment by thesamim** [comment]
> I really like what Ahcorn proposes, and I ask if I can here the links in obsidian to ticktick and necessary or a feature?

They are. Imagine having thousands of tasks. Image those tasks being in files that don't necessarily reflect the list name in Ticktick. Imagine wanting to do something with one of those tasks. Finding that task in Ticktick again could be  a challenge.

**Comment by DIGITAL-DECAY** [comment]
As above, the plugin is great, and I appreciate what you're saying about the ease of finding where tasks relate to if you had an enormous number of tasks, but even so the option to NOT include links would be very much appreciated.
I don't have thousands of tasks, so the inclusion of links actually makes it much harder to parse the tasks I'm looking at (moreso in TickTick than Obsidian), and even though I can manually remove the links, it has to be done separately in both TickTick and Obsidian which kind of defeats the convenience of syncing tasks.

**Comment by Joezcool** [comment]
Throwing my two cents in to hopefully bump the feature, I'd also appreciate the option for the links to be either removed -- or more reasonably, moved to the description. Particularly in ticktick, where seeing 

> Taskname [Ticktick/Inbox.md]               #ticktick

 or similar for all my tasks in the app can be an eyeful for sure.  
 
I'll also agree with the idea that _in ticktick_, the tag ought to be #obsidian instead of #ticktick. Alternatively, if having the tags be consistent is important to the implementation, a more vague tag like #synched, #ots (obsidian-ticktick-synched), etc. or a user-specificed tag for both could work. (Having the tag be whatever client you use less makes sense, but if you use both at least semi frequently, which I imagine someone bothering to use the plugin in the first place may, this consistency would be valuable.)

Additionally, there's a world where if your _only_ tasks in obsidian are ticktick tasks, you should reasonably be able to forgo the tag entirely and even theoretically the inverse, but both of these are a bit more niche of a feature to request and significantly less pressing than the link title imo.

@thesamim Would you want help implementing this? I could put together a PR for these changes.

**Comment by thesamim** [comment]
> @thesamim Would you want help implementing this? I could put together a PR for these changes.

@Joezcool : Definitely would appreciate the help, but would ask that you hold off until  I get https://github.com/thesamim/TickTickSync/issues/97 completed. There are some major changes to date handling, which affect Task Parsing and it would make more sense to add the feature once that's complete.

**Comment by davetorbeck** [comment]
Hi, big fan of the plugin. I'm also of the persuasion that links are a bit too noisy to my eyes for TickTick daily usage.   Any chance this will be worked on next?

**Comment by thesamim** [comment]
> Hi, big fan of the plugin. I'm also of the persuasion that links are a bit too noisy to my eyes for TickTick daily usage. Any chance this will be worked on next?

In the middle of a pretty substantial code refactor right now. Note synchronization is next on the list. Then this.

**Comment by MohamedKelif** [comment]
Yes I agreed it’s visually cluttering for both TickTick and obsidian Side. It will be nice if users were allowed to disable the those features. I use my own tags on TickTick. A workaround I found out was to make the TickTick tag background Color white. That helped easy my eyes. 

### Features I would love to have are: TickTick Side

1. Allow user to edit TickTick link for placing it either on the title or description similar to other TickTick plugin. This can be done using pop out modal form. 

2. Allow users to disable tags from TickTick 

### Obsidian Side: 
1. Allow users to disable tags and TickTick IDs this can’t be filtered out by the task plugin. 

2. The link to redirect to ticktick is nice but it will be nice if it was within the task name. What do you think 🤔 

I have added so photo for reference from the other  TickTick plugin 

![Image](https://github.com/user-attachments/assets/5c24aa9b-a920-42e5-bb5b-32b646289e9c)

![Image](https://github.com/user-attachments/assets/a8796a26-c459-41fe-aa9b-c6ed3d5a4562)

![Image](https://github.com/user-attachments/assets/26e6a56a-a880-4a04-83e7-dc7072dded19)

**Comment by thesamim** [comment]
@davetorbeck, @martin-arg , @AHCorn . @MohamedKelif :
Pardon the delay in response. 

Your suggestions and recommendations are noted. 

I am working on the next release. The primary focus of the next release is a major refactor and synchronization of notes. When I (finally) get that done, I will look into ways of accommodating the requests.

Stay tuned.

**Comment by thesamim** [comment]
Link placement is now configurable.

If default tag configuration is still a need, please open a new issue.
